language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/SubmitCommandsRequest.java

```diff
@@ -33,16 +38,35 @@ public SubmitCommandsRequest(@NonNull String workflowId, @NonNull String applica
                                  @NonNull String commandId, @NonNull String party, @NonNull Optional<Instant> minLedgerTimeAbsolute,
                                  @NonNull Optional<Duration> minLedgerTimeRelative, @NonNull Optional<Duration> deduplicationTime,
                                  @NonNull List<@NonNull Command> commands) {
+        this(
+                workflowId, applicationId, commandId, list(party), list(),
+                minLedgerTimeAbsolute, minLedgerTimeRelative, deduplicationTime,
+                commands);
+    }
+
+    public SubmitCommandsRequest(
+            @NonNull String workflowId,
+            @NonNull String applicationId,
+            @NonNull String commandId,
+            @NonNull List<@NonNull String> actAs,
+            @NonNull List<@NonNull String> readAs,
+            @NonNull Optional<Instant> minLedgerTimeAbsolute,
+            @NonNull Optional<Duration> minLedgerTimeRelative,
+            @NonNull Optional<Duration> deduplicationTime,
+            @NonNull List<@NonNull Command> commands) {
         this.workflowId = workflowId;
         this.applicationId = applicationId;
         this.commandId = commandId;
-        this.party = party;
+        this.party = actAs.get(0);
```

Comment:
Maybe a good idea to throw a better error message than just an index out of bounds here?